src/e2e.spec.ts

```diff
@@ -114,72 +116,80 @@ describe('e2e', () => {
-
-      @restate.service<UserServiceApi>()
-      class UserController implements UserService {
-        constructor(private readonly ctx: RestateContext) {}
-
-        @restate.method()
-        async create(username: string): Promise<User> {
-          return new User(username);
-        }
-      }
-
-      const app = createTestingApp({
-        imports: [new RestateModule({ port: 9081 })],
-        controllers: [UserController],
-      });
-      void app.startServer();
+      constructor(readonly username: string) {}
+    }
 
-      {
-        const response = await createDeployment(9081);
-        expect(await response.json()).toMatchObject({
-          services: [
-            {
-              name: 'user',
-              instance_type: 'Unkeyed',
-              methods: [
-                {
-                  input_type: 'RpcRequest',
-                  name: 'create',
-                  output_type: 'RpcResponse',
-                },
-              ],
-            },
-          ],
-        });
-      }
+    interface UserService {
+      create(username: string): Promise<User>;
+    }
 
-      const client = new RestateClient('http://0.0.0.0:8080');
+    type UserServiceApi = RestateService<'user', UserService>;
 
-      const user = client.service<UserServiceApi>();
+    @restate.service<UserServiceApi>()
+    class UserController implements UserService {
+      constructor(private readonly ctx: RestateServiceContext) {}
 
-      {
-        const result = await client.rpc(user.create('Test'));
-        expect(result).toBeInstanceOf(User);
-        expect(result).toMatchObject({
-          id: expect.any(String),
-          username: 'Test',
-        });
+      @restate.method()
+      async create(username: string): Promise<User> {
+        return new User(username);
       }
+    }
 
-      {
-        const result = await client.send(user.create('Test'));
-        expect(result.id).toMatch(/^inv_/);
-      }
+    const app = createTestingApp({
+      imports: [new RestateModule({ port: 9081 })],
+      controllers: [UserController],
     });
+    void app.startServer();
+
+    {
+      const response = await createDeployment(9081);
+      // expect(await response.json()).toMatchObject({
+      //   services: [
+      //     {
+      //       name: 'user',
+      //       instance_type: 'Unkeyed',
+      //       methods: [
+      //         {
+      //           input_type: 'RpcRequest',
+      //           name: 'create',
+      //           output_type: 'RpcResponse',
+      //         },
+      //       ],
+      //     },
+      //   ],
+      // });
+    }
+
+    const client = new RestateClient({ url: 'http://0.0.0.0:8080' });
+
+    const user = client.service<UserServiceApi>();
+
+    {
+      const user = new User('Test');
+      const serialize = getBSONSerializer<User>();
+      const deserialize = getBSONDeserializer<User>();
+      expect(deserialize(serialize(user))).toMatchObject(user);
+    }
```

Comment:
Doesn't work.
I get the following error `AssertionError: expected User{ username: undefined, …(1) } to match object User{ username: 'Test', …(1) }`

Reply:
Workaround is to add `public` to `username`.